### ISSUE TITLE
Last frame of media source flashes before starting video playback

### ISSUE THREAD

**Issue by mercantigo** [obsproject/obs-studio#4667]
### Operating System Info

Windows 10

### OBS Studio Version

27.0.0-rc3

### OBS Studio Log URL

https://obsproject.com/logs/BDsWd78G9u6yWpdA

### Expected Behavior

Play previously video starting with first frame

### Current Behavior

First plays last frame and then starts playing from first frame

### Steps to Reproduce

Add 'Media source'
Select a local video file (tested with quicktime and transparency and also h264 mp4) 
Configure playback with the following settings:
 [ ] loop
 [x]  restart playback when source becomes active
 [x] or [ ] use hardware enc..
 [ ] show noth...
 [x] or [ ] close file when in...

 Switch between scenes and the problem appears most of the time, problem is most consistent around lower framerates (25 fps) bu tstill noticeable around higher framerates (50 fps)

### Anything else we should know?

The bug was firstly reported in https://github.com/obsproject/obs-studio/issues/2906, but it was solved. Now, the bug returned...

**Comment by RytoEX** [comment]
According to the comments on the Issue that you linked, this returned between 26.0.0 and 26.1.1.

> I seem to be having this issue again with 26.1.1

> I can confirm this is also happening to me on 26.1.1.

If that's the case, then this isn't specifically new to the OBS Studio 27 Release Candidates.  Can you reproduce this on the 26.x releases?

**Comment by mercantigo** [comment]
> According to the comments on the Issue that you linked, this returned between 26.0.0 and 26.1.1.
> 
> > I seem to be having this issue again with 26.1.1
> 
> > I can confirm this is also happening to me on 26.1.1.
> 
> If that's the case, then this isn't specifically new to the OBS Studio 27 Release Candidates. Can you reproduce this on the 26.x releases?

I managed to reproduce on 26.x release.
I had an older version before (maybe 25.x?) and, on then, the issue didnt happen.

**Comment by RoyceRemix** [comment]
So I've also had this issue in v27, but I've come up with a workaround solution utilizing the new Source Transition feature!

Make an Image Source of the last frame of your video and set the Show Transition (stinger) for it to your video file.  This way when the transition ends, the video completely unloads and you have a seamless transition into the last frame as a still image.  Done!

**Comment by SimonGZ** [comment]
Noting that this continues to be a problem in OBS 27.2.1 on Mac. This old forum post does a really nice job of visualizing the issue: https://obsproject.com/forum/threads/media-source-rendering-the-last-frame-at-the-beginning.110180/

**Comment by DrewSunn** [comment]
Problem persists on 28.0.1

**Comment by tt2468** [comment]
@mercantigo Are you still able to reproduce this on 29.0.2?

**Comment by gravelfreeman** [comment]
Unfortunately we changed our After Effects template and we're not using folders in OBS anymore so I can't tell.

**Comment by SimonGZ** [comment]
I have tested this problem on 30.0.0 and unfortunately the problem remains (at least on my M1 Pro Macbook). I was hoping that #3206 would have fixed the issue but no, I still see the last frame of whatever video just played whenever I enter a slide with a media source set to "Restart playback when source becomes active"

**Comment by CSMBY** [comment]
Before version 30, this bug was reproduced at 60 FPS, and from version 30+ this bug is reproduced at 30 FPS. That's why I always used 30 FPS. But now that doesn't help either. A very disgusting bug, it greatly affects the quality of the output video. I would even say that it becomes a shame for such quality of the video ((((

**Comment by CSMBY** [comment]
> @mercantigo Are you still able to reproduce this on 29.0.2?

The bug is reproduced on all versions 29 and all versions 30. The most basic way to reproduce:
1) Make a transparent stinger 1 second long, set the transition to 500 ms
2) Make two scenes where in each a video file is played with a restart option
2) Constantly switching between two scenes

**Comment by codemann8** [comment]
Is there any update to this issue? I've been monitoring this issue since February and am REALLY looking forward to this being fixed, but it doesn't seem to have went anywhere recently.

**Comment by parish-of-brentford** [comment]
Bug still exists in 31.0.0. The first time the media source is played it works perfectly but subsequent replays (when the source becomes active) start by flashing up the last frame from when the source was last stopped. Seems like the frame buffer is not being cleared down when playback is stopped/finishes.

**Comment by Poodmund** [comment]
I've had a bash at resolving this and it seems to be working in my debug build. I have absolutely no idea on the ramifications of what I've done however, so anyone who could review would be greatly appreciated: https://github.com/Poodmund/obs-studio/commit/4986749ff5b693fc21d24d504b4ec2c6cf109750.

**Comment by SimonGZ** [comment]
Very exciting. I will try to get this running on my machine to at least offer basic feedback of “yes it worked” or “no it didn’t.”

**Comment by SimonGZ** [comment]
I have tried out @Poodmund's change and in my limited testing, it seems like a huge improvement. I don't see the last frame popping in anymore.

In my ideal world, the first frame would preload in some way because I can still see a fraction of a second where the scene loads and then the media source pops in afterward, but it's much less distracting now because it's starting at the beginning rather than flashing the last frame of the video then jerking back to the beginning.

Thank you for your work on this and I hope it's a change that can be approved and added for a future release.

**Comment by contentfree** [comment]
Any chance the PR associated with this will be merged into the next release soon?

**Comment by codemann8** [comment]
@tt2468 See above comment

**Comment by SpokeLee** [comment]
32.0.1 still has this. Is there an idiots guide to installing a commit from source? I really would like to try @Poodmund's version.
